crates/fluvio-benchmark/src/benchmark_config/benchmark_matrix.rs

```diff
@@ -0,0 +1,189 @@
+use std::time::Duration;
+use fluvio::Compression;
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+
+use super::benchmark_settings::{BenchmarkSettings, BenchmarkBuilder, CrossIterate};
+/// Key used by AllShareSameKey
+pub const SHARED_KEY: &str = "SHARED_KEY";
+
+/// DEFAULT CONFIG DIR
+pub const DEFAULT_CONFIG_DIR: &str = "crates/fluvio-benchmark/benches";
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct SharedSettings {
+    pub matrix_name: String,
+    pub num_samples: usize,
+    pub millis_between_samples: u64,
+    pub worker_timeout_seconds: u64,
+}
+
+/// A BenchmarkMatrix contains a collection of settings and dimensions.
+/// Iterating over a BenchmarkMatrix produces a BenchmarkSettings for every possible combination of values in the matrix.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct BenchmarkMatrix {
+    pub shared_settings: SharedSettings,
+
+    // Dimensions
+    pub num_records_per_producer_worker_per_batch: Vec<u64>,
+    pub producer_batch_size: Vec<u64>,
+    pub producer_queue_size: Vec<u64>,
```

Comment:
it would be probably better to nest this instead of trying to flatten everything. so 
```
pub struct BenchmarkMatrix {
       pub producer_config,
       pub consume_config

}
```

Reply:
That's a good point, I nested this. This matrix becomes a collection of intermediate `builders` and finally `config` for a single run. `builder` and `config` are internal and also still flattened. Do you have a strong opinion one way or other about unflattening `builder` and `config`

Reply:
builder and config have been merged, flat structure is better for builder pattern i think.